homeassistant/components/sma/sensor.py

```diff
@@ -84,7 +84,11 @@ def __init__(
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return self._sensor.name
+        name_prefix = "SMA"
+        if self._attr_device_info is not None:
+            name_prefix = str(self._attr_device_info.get("name", "SMA"))
```

Comment:
```suggestion
        if self._attr_device_info is None or (
            name_prefix := self._attr_device_info.get("name")
        ):
            name_prefix = "SMA"
```

Reply:
Any advantage of using
```python
not (  name_prefix := self._attr_device_info.get("name")  )
```

over 
```python
(  name_prefix := self._attr_device_info.get("name")  ) is not None
```


Could we also do this than?
```python
        if not self._attr_device_info or not (
            name_prefix := self._attr_device_info.get("name")
        ):
```

Reply:
It catches the case if `name` would be an empty string... 🤷

Reply:
Ah yes :)!